sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/util/CharVarcharCodegenUtils.java

```diff
@@ -54,8 +51,6 @@ public static UTF8String varcharTypeWriteSideCheck(UTF8String inputStr, int limi
       if (numChars <= limit) {
         return inputStr;
       } else {
-        // Trailing spaces do not count in the length check. We need to retain the trailing spaces
-        // (truncate to length N), as there is no read-time padding for varchar type.
         int maxAllowedNumTailSpaces = numChars - limit;
```

Comment:
ditto

Reply:
we can probably create a private method for the logic below to save duplicated code.

Reply:
sgtm